activesupport/test/message_verifier_test.rb

```diff
@@ -24,6 +24,7 @@ def test_valid_message
     data, hash = @verifier.generate(@data).split("--")
     assert !@verifier.valid_message?(nil)
     assert !@verifier.valid_message?("")
+    assert !@verifier.valid_message?("\xff")
```

Comment:
I'm not sure if I follow. Instead of returning `false` on this case, I would `raise` something like `InvalidEncoding`? If so, wouldn't `verify` end up raising the same instead of `InvalidSignature`?